### PR TITLE
:bug: fixup request cancellation for v0.6.5

### PR DESCRIPTION
**Pull request by joerunde**
Compatibility fix for vllm v0.6.5

## Description

vLLM has removed the cancellation polling in iterate_with_cancellation and merge_async_iterators, moving it up to the http request handler. This PR removes those integrations, and adds logging integration when requests are cancelled.

It turns out the async grpc server implementation will cancel the task running the handler when the context cancels, so there is no need to poll or implement our own cancellation handling anyway. The LLM engine implementations in vLLM already also catch `asyncio.CancelledError` and abort the work in the engine for us.

TL;DR: We didn't need this anyway

## How Has This Been Tested?
Manual verification of logs and that work is removed from the engine's queues

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by fialhocoelho** [comment]
LGTM

**Review by fialhocoelho — APPROVED** [review]
LGTM

**Comment by joerunde** [comment]
@fialhocoelho @dtrifiro whoops, looks like I never added this to the merge queue? :(
